oauth.go

```diff
@@ -350,6 +359,115 @@ func (x *OAuth) HttpHandlerOAuthTest(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Provides support for username and password login to MSAAD
+// App to app authentication.
+// This is NOT recommended for normal user access but for trusted applications that have their own user credentials in MSAAD
+// AND is linked to the same tenant ID under which auth operates.
+
+func (x *OAuth) OAuthLoginUsernamePassword(r *http.Request, username string, password string) error {
+	// Microsoft Azure Active Directory is the only provider we've needed to implement so far
+
+	/* TODO : We refer to the "provider" by name in the config (e.g. emerge), but in the database, we refer to it by
+	type alone. In essence the store does not have any information on WHICH provider will be used to authenticate against,
+	should there be more than one of the same type.
+	*/
+
+	var provider *ConfigOAuthProvider
+	for _, p := range x.Config.Providers {
+		if p.Type == OAuthProviderMSAAD {
+			provider = p
+		}
+	}
+	if provider == nil {
+		errorMsg := fmt.Sprintf("OAuth provider '%v' not configured", OAuthProviderMSAAD)
+		return errors.New(errorMsg)
+	}
+
+	//client_id:54358a07-89cb-40b4-b99f-5a981c627f58
+	//scope:user.read openid profile
+	//client_secret:
+	//grant_type:password
+	//username:SA_PA_ASSETCORE@awemerge.co.za
+	//password:
+
+	params := map[string]string{
+		"client_id":     provider.ClientID,
+		"scope":         provider.Scope,
+		"client_secret": url.QueryEscape(provider.ClientSecret),
+		"grant_type":    "password",
+		"username":      username,
+		"password":      password,
+	}
+
+	// make the call to msaad
+	resp, err := http.DefaultClient.Post(provider.TokenURL, "application/x-www-form-urlencoded", strings.NewReader(buildPOSTBodyForm(params)))
+	if err != nil {
+		err1 := fmt.Errorf("Error acquiring access token: %w", err)
+		return err1
```

Comment:
Just return the `fmt.Errorf(...`. No need to have a temporary variable.

Reply:
Classic case of deciding against loggin it as well...and leaving the intermediate variable in place. Will remove.